clouddrift/analysis.py

```diff
@@ -1198,3 +1208,239 @@ def unpack_ragged(
+    # )
+    longitude_new, latitude_new = cartesian_to_spherical(
+        x - np.real(wx), y - np.real(wy), z - np.real(wz)
+    )
+
+    # convert transforms to horizontal displacements on tangent plane
+    wxh, wyh = cartesian_to_tangentplane(wx, wy, wz, longitude_new, latitude_new)
+
+    # rotary wavelet transforms to select inertial component; need to divide by sqrt(2)
+    wp = (wxh + 1j * wyh) / np.sqrt(2)
+    wn = (wxh - 1j * wyh) / np.sqrt(2)
+
+    # find the values of radian_frequency/dt that most closely match cor_freq
+    # is there a better way to do this?
+    frequency_bins = [
+        np.argmin(np.abs(cor_freq[i] - radian_frequency / time_step))
+        for i in np.arange(0, data_length)
+    ]
+
+    # get the transform at the inertial and "anti-inertial" frequencies
+    # extract the values of wp and wn at the calculated index as a function of time
+    # positive is anticyclonic (inertial) in the southern hemisphere
+    # negative is anticyclonic (inertial) in the northern hemisphere
+    wp = wp[frequency_bins, list(range(0, data_length))]
+    wn = wn[frequency_bins, list(range(0, data_length))]
+
+    # need an index of northen latitude points and index of
+    # southern latitude points
+    north = latitude >= 0
+
+    # initialize the zonal and meridional components of inertial displacements
+    wxhat = np.zeros_like(latitude).astype(np.complex64)
+    wyhat = np.zeros_like(latitude).astype(np.complex64)
+    # equations are x+ = 0.5*(z+ + z-) and y+ = -0.5*1j*(z+ - z-)?
+    if any(north):
+        wxhat[north] = wn[north] / np.sqrt(2)
+        wyhat[north] = 1j * wn[north] / np.sqrt(2)
+    if any(~north):
+        wxhat[~north] = wp[~north] / np.sqrt(2)
+        wyhat[~north] = -1j * wp[~north] / np.sqrt(2)
+
+    # inertial displacement in kilometers
+    xhat = 1e-3 * np.real(wxhat)
+    yhat = 1e-3 * np.real(wyhat)
+    xy = xhat + 1j * yhat
+
+    # longitude and latitude after removing oscillations
+    longitude_cor, latitude_cor = corrected_positions_from_displacements(
+        longitude, latitude, xhat, yhat
+    )
+
+    return xy, longitude_cor, latitude_cor
+
+
+def corrected_positions_from_displacements(
+    longitude: Union[float, np.ndarray],
+    latitude: Union[float, np.ndarray],
+    x: Union[float, np.ndarray],
+    y: Union[float, np.ndarray],
+) -> Union[Tuple[float], Tuple[np.ndarray]]:
+    """
+    Return the longitude and latitude along a trajectory on the spherical Earth
+    after correcting for zonal and meridional displacements x and y in kilometers.
+
+    This is applicable as an example when one seeks to correct a trajectory for
+    horizontal oscillations.
+
+    Parameters
+    ----------
+    longitude : float or np.ndarray
+        Longitude in degrees.
+    latitude : float or np.ndarray
+        Latitude in degrees.
+    x : float or np.ndarray
+        Zonal displacement in kilometers.
+    y : float or np.ndarray
+        Meridional displacement in kilometers.
```

Comment:
Do we have to put this in km? I would keep everything in SI if possible. 

If we keep km, I would create a variable `METERS_PER_KM` to perform the conversion instead of using 1e-3. Or, just a `EARTH_RADIUS_KM` defined once at the top.

Reply:
I agree with the SI argument but I am curious of the following: why is it that having an obvious variable `METERS_PER_KM` is better than 1e-3?

Reply:
Because you have to guess what is happening to have to multiply by 1e-3. I had to go to the docstring and saw that it was in km. In general, we should avoid multiplying by any numbers and create either a variable to explain what it does. Exception to very obvious things like `1/2 * (a+b)`).

Reply:
That makes sense. However I think we should stick to SI as a principle and return meters ...